### PR TITLE
IA-3463 Add group check for createApp API

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3463

Security team wants to make it so that creating CUSTOM app is only allowed for a specific group

I tested locally with both a user in the group and a user isn't in the group with local leo against dev everything else.

Screen Shot 2022-06-02 at 3 29 22 PM

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala [hunk 1]
```diff
@@ -325,7 +326,10 @@ object Config {
       SamAuthProviderConfig(
         config.getOrElse("notebookAuthCacheEnabled", true),
         config.getAs[Int]("notebookAuthCacheMaxSize").getOrElse(1000),
-        config.getAs[FiniteDuration]("notebookAuthCacheExpiryTime").getOrElse(15 minutes)
+        config.getAs[FiniteDuration]("notebookAuthCacheExpiryTime").getOrElse(15 minutes),
+        config
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala:330]: Minor - could do `config.as[GroupName]("customAppCreationAllowedGroup")` to not get an Option?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala [hunk 2]
```diff
@@ -383,6 +383,31 @@ class HttpSamDAO[F[_]](httpClient: Client[F],
...
+          method = Method.GET,
+          uri = config.samUri.withPath(Uri.Path.unsafeFromString(s"/api/groups/v1/${groupName.asString}/member")),
+          headers = Headers(leoAuth)
+        )
+      )(onError)
+      res <- if (members.contains(workbenchEmail))
+        F.pure(true)
+      else
+        for {
+          admins <- httpClient.expectOr[List[WorkbenchEmail]](
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala:402]: Would be nice if we didn't have to make a Sam call for each policy but I don't see a way in the Sam API

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala:402]: I know...wish Sam api supports that

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala [hunk 3]
```diff
@@ -201,4 +201,6 @@ trait LeoAuthProvider[F[_]] {
 
   // Get user info from Sam. If petOrUserInfo is a pet SA, Sam will return it's associated user account info
   def lookupOriginatingUserEmail[R](petOrUserInfo: UserInfo)(implicit ev: Ask[F, TraceId]): F[WorkbenchEmail]
+
+  def isAllowed(userEmail: WorkbenchEmail)(implicit ev: Ask[F, TraceId]): F[Boolean]
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala]: A slightly more descriptive name would be nice? :) Maybe `isCustomAppAllowed`

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala (hunk 1 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala:330]: best I can see is `getOrElse`...guess I can do that

**Review by rtitle — APPROVED** [review]
Looks correct to me. Remember that the group needs to be created in each env (and Leo SA added)

**Pull request merged**